nservicebus/azure-servicebus/topologies/index.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Azure Service Bus Topologies
+summary: Azure Service Bus Topologies
+tags:
+- Azure
+- Cloud
+redirects:
+related:
+ - samples/azure/azure-service-bus
+reviewed: 2016-04-12
+---
+
+Messaging topology is an arrangement of the various messaging entities.
+Azure Service Bus transport operates on a topology created on the broker. Messaging entities participating in topology are queues, topics, subscriptions, and rules. Combined in a specific way they form a topology. 
+
+Topology handles message exchange between endpoints by creating and configuring Azure Service Bus entities.  
+
+## Version 6 and below
+
+Version 6 and below of the transport has always supported a single default topology out-of-the-box. It was an implicit implementation of what was defined as `EndpointOrientedTopology` in version 7 and up.
+
+## Version 7 and up
+
+With version 7 of the transport, topology concept was introduced. There are two topologies:
+
+1. `EndpointOrientedTopology`
+1. `ForwardingTopology`
+
+`EndpointOrientedTopology` is the default topology. It's fully backwards compatible with version 6 and below of the transport.`ForwardingTopology` was introduced to take advantage of the broker nature of Azure Service Bus and leverage the native capabilities. This makes `ForwardingTopology` a much better option for new projects. 
+
+Both topologies create input queue per endpoint and implement pub/sub mechanism. Significant difference between the two is how pub/sub mechanism is implemented.
+
+### Endpoint Oriented Topology
+
+`EndpointOrientedTopology` relies heavily on the endpoint information. This information is used to build the pub/sub mechanism. Each publishing endpoint creates a topic using the convention of `<endpoint name>.events`. Events are published to the topic. All the subscribing endpoints subscribe to the publisher topic by creating subscription per event using the convention of `<subscriber endpoint name>.<event name>`. Each subscription has a rule to select a specific event, `event`. 
```

Comment:
In the `EndpointOrientedTopology` each publishing endpoint creates a topic called `<publishing_endpoint_name>.events`. The subscribing endpoints subscribe to the topic, by creating a subscription for a particular even type called `<subscriber_endpoint_name>.<event_type_name>`. Note that each subscription has a rule, used to filter a specific event type. 

When a publisher raises an event, it is published to the publisher topic. Then the subscription entities filter events using the associated rules. Finally, the event is sent to the subscribing endpoints input queue.